### PR TITLE
Update and complete German translation and fix mistakes

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR updates the German translation, completes it (100% translated) and fixes some grammatical and typographical mistakes.

### CODE REVIEW

File: po/de/minetest.po [hunk 1]
```diff
@@ -106,11 +106,13 @@ msgstr "Warnung: Die minimale Testversion ist für Entwickler gedacht."
...
 msgid "Download a subgame, such as minetest_game, from minetest.net"
-msgstr "Andere Spiele (wie minetest_game) können von minetest.net heruntergeladen werden"
+msgstr ""
+"Andere Spiele (wie minetest_game) können von minetest.net heruntergeladen "
+"werden"
 
 #: builtin/mainmenu/dlg_create_world.lua:97
 msgid "A world named \"$1\" already exists"
-msgstr "Eine Welt namens \"$1\" existiert bereits"
+msgstr "Eine Welt namens »$1« existiert bereits"
```
Comment (BlockMen) [inline po/de/minetest.po]: "Blabla" ist eindeutiger

File: po/de/minetest.po [hunk 2]
```diff
@@ -164,19 +166,20 @@ msgid ""
 "Install Mod: unsupported filetype \"$1\" or broken archive"
 msgstr ""
 "\n"
-"Mod installieren: Nicht unterstützter Dateityp \"$1\" oder fehlerhaftes Archiv"
+"Mod installieren: Nicht unterstützter Dateityp »$1« oder kaputtes Archiv"
```
Comment (BlockMen) [inline po/de/minetest.po]: fehlerhaft oder defekt. kaputt ist zu umangssprachlich

File: po/de/minetest.po [hunk 3]
```diff
@@ -248,15 +251,15 @@ msgstr "Installierte Mods:"
 
 #: builtin/mainmenu/tab_mods.lua:39
 msgid "Online mod repository"
-msgstr "Online Mod Speicher"
+msgstr "Online-Mod-Speicher"
```
Comment (BlockMen) [inline po/de/minetest.po]: (Online) Mod Archiv
Speicher ist eh eine schlechte übersetzung

**Comment by BlockMen** [comment]
Please update, especially revert the dots "..."
Concerning the hyphenation, either "Mod Informationen" or "Mod-Informationen", but not "Modinformationen".

**Comment by Wuzzy2** [comment]
Oh, BlockMen …

* “Mod Informationen”, “Online Mod Archiv”, “Mod Archiv”: All wrong, this is the Deppenleerzeichen. https://de.wikipedia.org/wiki/Leerzeichen_in_Komposita
* You might however argue for “Mod-Informationen” over “Modinformationen”. But at the end of the day, this is not much of a difference.
* Using the “ASCII quote” as quotation marks: Sorry, this is simply wrong in German, and your personal preference is not going to change that. How is clearness even an argument here? Either „…“ or »…«, but not "…". The only valid counter-argument would be valid character encoding or font concerns.
* What's wrong with the ellipsis character, anyways? Are there *real* (not just maybe) encoding or font problems?

Anyways, this PR is now really outdated and messy. A couple of *new* strings have been added lately. So I guess I just throw this PR away and start over, considering the *reasonable* comments this time.

**Pull request closed without merging**